src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs

```diff
@@ -921,5 +922,199 @@ class C1
                 Assert.Empty(symbolInfo.CandidateSymbols);
             }
         }
+
+        [Fact]
+        [WorkItem(30906, "https://github.com/dotnet/roslyn/issues/30906")]
+        public void NullableTupleWithTuplePattern_01()
+        {
+            var source = @"using System;
+class C {
+    static (int, int)? Get(int i)
```

Comment:
Worth testing nested positions?
```
(int, (int, int)?)  Get()
Get() is var (x, (y, z))
```

Reply:
Yes, especially since testing for that revealed a nasty bug (now fixed)

---
In reply to: [230521763](https://github.com/dotnet/roslyn/pull/30935#discussion_r230521763) [](ancestors = 230521763)